build/build.go

```diff
@@ -476,7 +476,7 @@ func BuildWithResultHandler(ctx context.Context, nodes []builder.Node, opt map[s
 						rr.ExporterResponse[k] = string(v)
 					}
 					rr.ExporterResponse["buildx.build.ref"] = buildRef
-					if opt.WithProvenanceResponse && node.Driver.HistoryAPISupported(ctx) {
+					if !so.Internal && opt.WithProvenanceResponse && node.Driver.HistoryAPISupported(ctx) {
```

Comment:
Why do we need `!so.Internal` if `WithProvenanceResponse` is already false.

Reply:
This was for extra safety as build record does not exist for internal builds but not needed as internal solve opt can't be set with current api.